05_test.sh

```diff
@@ -131,6 +131,24 @@ check_bmh_status(){
   return "$((FAILS-FAILS_CHECK))"
 }
 
+#
+# Convert an ip address into an int for comparison
+#
+# Inputs:
+# - IP address (X.X.X.X)
+#
+ip_to_int(){
+  local IP="${1}"
+  local INT_IP=0
+  local iteration=3
+  for i in $(echo "$IP" | tr '.' ' '); do
+    ((INT_IP=INT_IP+(i*(2**(iteration*8)))))
+    ((iteration=iteration-1))
+  done
+  echo "${INT_IP}"
+}
```

Comment:
I wonder if bash is the best tool for the job here - I can see this works but it's not super clear?

How about using python for the IP comparison, e.g 

```
# Usage ip_in_range ip start end
ip_in_range() {
python -c "
import ipaddress
import sys
if int(ipaddress.IPv4Address(u'"$2"')) <= int(ipaddress.IPv4Address(u'"$1"')) <= int(ipaddress.IPv4Address(u'"$3"')):
  sys.exit(0)
else:
  sys.exit(1)
"
}

DHCP_START="172.22.0.100"
DHCP_END="172.22.0.200"
IP="172.22.0.150"
if ip_in_range $IP $DHCP_START $DHCP_END; then
    echo "OK"
else
    echo "FAIL"
fi
```